### ISSUE TITLE
Preventing step incrementation on close in controlled mode causes overlay to stay behind or stops tutorial from closing

### ISSUE THREAD

**Issue by rgembalik** [gilbarbara/react-joyride#1077]
## 🐛 Bug Report

I want to treat the close button as something that hides the tutorial but doesn't change the current step. (Hide tutorial, but keep beacons etc).

## To Reproduce

Based on the controlled example on CodeSandbox ([`src/Controlled/index.tsx`](https://codesandbox.io/p/devbox/bold-joana-ntxxm4?file=%2Fsrc%2FControlled%2Findex.tsx%3A62%2C1))

1. In the definition of the first step, set `disableOverlayClose` and `hideCloseButton` to `false`
2. In `handleJoyrideCallback` add `if(action === ACTIONS.CLOSE) return;` - in case user closes tutorial with `x` button, don't change step index.
3. Run the example and go to `/controlled` page
4. Click `x` button on the first step - first issue occurs, tutorial doesn't close
5. Open menu (causing navigation to the next step)
6. Click `x` button  - second issue occurs, tutorial closes, but overlay stays behind

## Expected behavior

With beacon:
The tutorial closes, beacon stays in place, overlay disappears.

Without beacon:
The tutorial closes, beacon stays in place, overlay disappears.

## Link to repl or repo (highly encouraged)

`/controlled` page in the following sandbox:

https://codesandbox.io/p/devbox/github/gilbarbara/react-joyride-demo/tree/main/

## Run `npx envinfo --system --binaries --npmPackages react-joyride`

Paste the results here:

```bash
  System:
    OS: Windows 11 10.0.22631
    CPU: (16) x64 13th Gen Intel(R) Core(TM) i7-1360P
    Memory: 11.87 GB / 31.71 GB
  Binaries:
    Node: 20.12.2 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.22 - C:\Program Files\nodejs\yarn.CMD
    npm: 10.5.0 - C:\Program Files\nodejs\npm.CMD
  npmPackages:
    react-joyride: ^2.8.2 => 2.8.2
```

**Comment by gilbarbara** [comment]
Hey @rgembalik 

> Based on the controlled example on CodeSandbox ([`src/Controlled/index.tsx`](https://codesandbox.io/p/devbox/bold-joana-ntxxm4?file=%2Fsrc%2FControlled%2Findex.tsx%3A62%2C1))
> 
> 1. In the definition of the first step, set `disableOverlayClose` and `hideCloseButton` to `false`
> 2. In `handleJoyrideCallback` add `if(action === ACTIONS.CLOSE) return;` - in case user closes tutorial with `x` button, don't change step index.
> 3. Run the example and go to `/controlled` page
> 4. Click `x` button on the first step - first issue occurs, tutorial doesn't close
> 5. Open menu (causing navigation to the next step)
> 6. Click `x` button  - second issue occurs, tutorial closes, but overlay stays behind

4 - Since the step has a `disableBeacon,` there is no fallback when you click the close button.
This isn't a bug. 

6 - I fixed this in v2.9.0

**Comment by rgembalik** [comment]
> 4 - Since the step has a disableBeacon, there is no fallback when you click the close button.
> This isn't a bug.
Understood. Yeah, I missed that.  It felt like this should close the tutorial with the ability to resume it programmatically (e.g. dedicated UI button), but it's already something that `run: false/true` can do.

> 6 - I fixed this in v2.9.0
Thank you!
